File: scholarship-web/src/constants/resultCode.ts
/**
 * 响应状态码常量
 */

/**
 * 成功
 */
export const SUCCESS = 200

/**
 * 客户端错误
 */
export const BAD_REQUEST = 400
export const UNAUTHORIZED = 401
export const FORBIDDEN = 403
export const NOT_FOUND = 404
export const METHOD_NOT_ALLOWED = 405
export const CONFLICT = 409
export const UNPROCESSABLE_ENTITY = 422
export const TOO_MANY_REQUESTS = 429

/**
 * 服务端错误
 */
export const INTERNAL_SERVER_ERROR = 500
export const BAD_GATEWAY = 502
export const SERVICE_UNAVAILABLE = 503
export const GATEWAY_TIMEOUT = 504

/**
 * Token 相关错误码
 */
export const TOKEN_EXPIRED = 6101
export const TOKEN_INVALID = 6102
export const TOKEN_BLACKLISTED = 6103

/**
 * 业务错误码
 */
export const VALIDATION_ERROR = 4000
export const DATA_NOT_FOUND = 4040
export const DUPLICATE_DATA = 4090
export const OPERATION_FAILED = 5000

/**
 * 错误消息映射
 */
export const ERROR_MESSAGES: Record<number, string> = {
  [SUCCESS]: '请求成功',
  [BAD_REQUEST]: '请求参数错误',
  [UNAUTHORIZED]: '未授权，请先登录',
  [FORBIDDEN]: '禁止访问',
  [NOT_FOUND]: '请求的资源不存在',
  [METHOD_NOT_ALLOWED]: '请求方法不允许',
  [CONFLICT]: '资源冲突',
  [UNPROCESSABLE_ENTITY]: '请求参数无法被处理',
  [TOO_MANY_REQUESTS]: '请求次数过多',
  [INTERNAL_SERVER_ERROR]: '服务器内部错误',
  [BAD_GATEWAY]: '网关错误',
  [SERVICE_UNAVAILABLE]: '服务不可用',
  [GATEWAY_TIMEOUT]: '网关超时',
  [TOKEN_EXPIRED]: 'Token 已过期',
  [TOKEN_INVALID]: 'Token 无效',
  [TOKEN_BLACKLISTED]: 'Token 已被加入黑名单',
  [VALIDATION_ERROR]: '数据验证失败',
  [DATA_NOT_FOUND]: '数据不存在',
  [DUPLICATE_DATA]: '数据重复',
  [OPERATION_FAILED]: '操作失败'
}

/**
 * 获取错误消息
 * @param code - 错误码
 * @param defaultMessage - 默认消息
 * @returns 错误消息
 */
export function getMessage(code: number, defaultMessage = '请求失败'): string {
  return ERROR_MESSAGES[code] || defaultMessage
}

/**
 * 判断是否为成功响应
 * @param code - 响应码
 * @returns 是否成功
 */
export function isSuccess(code: number): boolean {
  return code === SUCCESS
}

/**
 * 判断是否为 Token 相关错误
 * @param code - 响应码
 * @returns 是否为 Token 错误
 */
export function isTokenError(code: number): boolean {
  return [TOKEN_EXPIRED, TOKEN_INVALID, TOKEN_BLACKLISTED].includes(code)
}

export default {
  SUCCESS,
  BAD_REQUEST,
  UNAUTHORIZED,
  FORBIDDEN,
  NOT_FOUND,
  METHOD_NOT_ALLOWED,
  CONFLICT,
  UNPROCESSABLE_ENTITY,
  TOO_MANY_REQUESTS,
  INTERNAL_SERVER_ERROR,
  BAD_GATEWAY,
  SERVICE_UNAVAILABLE,
  GATEWAY_TIMEOUT,
  TOKEN_EXPIRED,
  TOKEN_INVALID,
  TOKEN_BLACKLISTED,
  VALIDATION_ERROR,
  DATA_NOT_FOUND,
  DUPLICATE_DATA,
  OPERATION_FAILED,
  ERROR_MESSAGES,
  getMessage,
  isSuccess,
  isTokenError
}

File: scholarship-admin/src/main/java/com/scholarship/common/result/ResultCode.java
package com.scholarship.common.result;

import lombok.AllArgsConstructor;
import lombok.Getter;

/**
 * 响应状态码枚举类
 * <p>
 * 定义系统中所有的响应状态码和对应的提示消息
 * </p>
 *
 * @author Scholarship Development Team
 * @version 1.0.0
 */
@Getter
@AllArgsConstructor
public enum ResultCode {

    // ========== 通用响应码 ==========

    /**
     * 操作成功
     */
    SUCCESS(200, "操作成功"),

    /**
     * 操作失败
     */
    ERROR(500, "操作失败"),

    // ========== 客户端错误 (4xx) ==========

    /**
     * 请求参数错误
     */
    BAD_REQUEST(400, "请求参数错误"),

    /**
     * 未授权，需要登录
     */
    UNAUTHORIZED(401, "未授权，请先登录"),

    /**
     * 禁止访问
     */
    FORBIDDEN(403, "禁止访问"),

    /**
     * 资源不存在
     */
    NOT_FOUND(404, "资源不存在"),

    /**
     * 请求方法不支持
     */
    METHOD_NOT_ALLOWED(405, "请求方法不支持"),

    /**
     * 请求超时
     */
    REQUEST_TIMEOUT(408, "请求超时"),

    /**
     * 请求实体过大
     */
    PAYLOAD_TOO_LARGE(413, "请求实体过大"),

    /**
     * 请求过于频繁
     */
    TOO_MANY_REQUESTS(429, "请求过于频繁，请稍后再试"),

    // ========== 服务端错误 (5xx) ==========

    /**
     * 服务器内部错误
     */
    INTERNAL_SERVER_ERROR(500, "服务器内部错误"),

    /**
     * 服务不可用
     */
    SERVICE_UNAVAILABLE(503, "服务不可用"),

    // ========== 业务错误码 (6xx) ==========

    /**
     * 用户名或密码错误
     */
    LOGIN_ERROR(6001, "用户名或密码错误"),

    /**
     * 用户已被禁用
     */
    USER_DISABLED(6002, "用户已被禁用"),

    /**
     * 用户不存在
     */
    USER_NOT_EXIST(6003, "用户不存在"),

    /**
     * 用户已存在
     */
    USER_ALREADY_EXIST(6004, "用户已存在"),

    /**
     * 密码错误
     */
    PASSWORD_ERROR(6005, "密码错误"),

    /**
     * 原密码错误
     */
    OLD_PASSWORD_ERROR(6006, "原密码错误"),

    /**
     * Token已过期
     */
    TOKEN_EXPIRED(6007, "Token已过期"),

    /**
     * Token无效
     */
    TOKEN_INVALID(6008, "Token无效"),

    // ========== 业务模块错误码 (601xx-609xx) ==========

    /**
     * 研究生信息不存在
     */
    STUDENT_NOT_EXIST(6011, "研究生信息不存在"),

    /**
     * 科研成果不存在
     */
    ACHIEVEMENT_NOT_EXIST(6012, "科研成果不存在"),

    /**
     * 科研成果已审核
     */
    ACHIEVEMENT_ALREADY_REVIEWED(6013, "科研成果已审核，不能重复操作"),

    /**
     * 评分规则不存在
     */
    SCORE_RULE_NOT_EXIST(6014, "评分规则不存在"),

    /**
     * 申请不存在
     */
    APPLICATION_NOT_EXIST(6015, "申请不存在"),

    /**
     * 申请已提交
     */
    APPLICATION_ALREADY_SUBMITTED(6016, "申请已提交，不能修改"),

    /**
     * 申请未提交
     */
    APPLICATION_NOT_SUBMITTED(6017, "申请未提交，不能审核"),

    /**
     * 评定批次不存在
     */
    BATCH_NOT_EXIST(6018, "评定批次不存在"),

    /**
     * 评定批次已结束
     */
    BATCH_ALREADY_ENDED(6019, "评定批次已结束"),

    /**
     * 评定批次未开始
     */
    BATCH_NOT_STARTED(6020, "评定批次未开始"),

    /**
     * 无权限操作
     */
    NO_PERMISSION(6021, "无权限操作");

    /**
     * 响应码
     */
    private final Integer code;

    /**
     * 响应消息
     */
    private final String message;
}
